app/pages/_app.tsx

```diff
@@ -47,19 +49,20 @@ export default class App extends NextApp<AppProps> {
     const {
       Component,
       router,
-      pageProps: {variables = {}, relayData}
+      pageProps: {variables = {}}
     } = this.props;
     const environment = createEnvironment(
-      relayData,
       JSON.stringify({
         queryID: Component.query
           ? getRequest(Component.query).params.name
           : undefined,
-        variables
+        variables: {...variables, ...router.query}
```

Comment:
Should this parameter be removed as well?

Reply:
need this one

Reply:
It's used in the client.ts fetchQuery function to determine whether the same query with the same variables is in the cache and to just use the cache in that case. This is also why mutations clear the cache.